File: android/app/src/main/java/com/example/demo/ui/components/RegionOverflowMenu.kt
package com.example.demo.ui.components

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.MoreVert
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.example.demo.data.RegionInfo
import com.example.demo.data.RegionPreferences

@Composable
fun RegionOverflowMenu(
    currentRegion: RegionInfo?,
    showRegionMenu: Boolean,
    onToggleMenu: () -> Unit,
    onSelectRegion: (String) -> Unit,
    onClearOverride: () -> Unit,
    modifier: Modifier = Modifier,
) {
    IconButton(
        onClick = onToggleMenu,
        modifier = modifier,
    ) {
        Icon(
            imageVector = Icons.Default.MoreVert,
            contentDescription = "More options",
        )
    }

    DropdownMenu(
        expanded = showRegionMenu,
        onDismissRequest = onToggleMenu,
    ) {
        Text(
            text =
                "Region: ${currentRegion?.name ?: "Unknown"}" +
                    if (currentRegion?.isOverride == true) " (Override)" else "",
            style = MaterialTheme.typography.labelMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
        )

        HorizontalDivider()

        RegionPreferences.AVAILABLE_REGIONS.forEach { region ->
            DropdownMenuItem(
                text = {
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Text(region.name)
                        if (currentRegion?.code == region.code) {
                            Spacer(modifier = Modifier.width(8.dp))
                            Icon(
                                imageVector = Icons.Default.Check,
                                contentDescription = "Selected",
                                modifier = Modifier.size(16.dp),
                                tint = MaterialTheme.colorScheme.primary,
                            )
                        }
                    }
                },
                onClick = { onSelectRegion(region.code) },
            )
        }

        if (currentRegion?.isOverride == true) {
            HorizontalDivider()
            DropdownMenuItem(
                text = { Text("Use Auto-detected") },
                onClick = onClearOverride,
            )
        }
    }
}

File: ios/ConfidenceDemoApp/UI/Components/RegionOverflowMenu.swift
import SwiftUI

struct RegionOverflowMenu: View {
    let currentRegion: RegionInfo?
    let showRegionMenu: Bool
    let onToggleMenu: () -> Void
    let onSelectRegion: (String) -> Void
    let onClearOverride: () -> Void

    @Environment(\.materialTheme) var theme

    var body: some View {
        VStack {
            Button(action: onToggleMenu) {
                Image(systemName: "ellipsis")
                    .foregroundColor(theme.onSurface)
                    .font(.title2)
                    .rotationEffect(.degrees(90))
            }

            if showRegionMenu {
                VStack(alignment: .leading, spacing: 0) {
                    // Region header
                    VStack(alignment: .leading, spacing: 4) {
                        Text("Region: \(currentRegion?.name ?? "Unknown")")
                            .font(.caption)
                            .foregroundColor(theme.onSurfaceVariant)

                        if currentRegion?.isOverride == true {
                            Text("(Override)")
                                .font(.caption2)
                                .foregroundColor(theme.onSurfaceVariant)
                        }
                    }
                    .padding(.horizontal, 16)
                    .padding(.vertical, 8)

                    Divider()

                    // Region options
                    ForEach(RegionPreferences.availableRegions) { region in
                        Button(action: {
                            onSelectRegion(region.code)
                        }) {
                            HStack {
                                Text(region.name)
                                    .foregroundColor(theme.onSurface)

                                Spacer()

                                if currentRegion?.code == region.code {
                                    Image(systemName: "checkmark")
                                        .foregroundColor(theme.primary)
                                        .font(.caption)
                                }
                            }
                            .padding(.horizontal, 16)
                            .padding(.vertical, 12)
                        }
                    }

                    if currentRegion?.isOverride == true {
                        Divider()

                        Button(action: onClearOverride) {
                            Text("Use Auto-detected")
                                .foregroundColor(theme.onSurface)
                                .padding(.horizontal, 16)
                                .padding(.vertical, 12)
                        }
                    }
                }
                .background(theme.surface)
                .clipShape(RoundedRectangle(cornerRadius: 8))
                .shadow(color: Color.black.opacity(0.1), radius: 4, x: 0, y: 2)
                .overlay(
                    RoundedRectangle(cornerRadius: 8)
                        .stroke(Color.gray.opacity(0.2), lineWidth: 1)
                )
                .zIndex(1)
            }
        }
    }
}